### ISSUE TITLE
amp-selector: optionally support (left/right)|(up/down) keyboard support for options

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#8613]
Add `keyboard-support` with  three modes:
1. `none`: No left/right key support.
2. `focus` (default): It only `focuses` the option when user navigates to it using left/right keys.
3. `select`: It `selects` the option when user navigates to it using left/right keys.

**Other requirements:**
Needs to cycle through options when reaching the end.

Also See https://github.com/ampproject/amp-by-example/pull/697#issuecomment-292188551

**Comment by ericlindley-g** [comment]
Sounds good — is this also a standard best practice for accessibility? (I'm assuming TAB would take the user out of the selector, to the next item on the page?)

**Comment by aghassemi** [comment]
Technically yes, it is recommended that if there is left/right keyboard support, tabs would no longer be navigating between the options BUT practically we should NOT do it that way. It breaks mobile screen readers ( as they don't see the options focusable anymore ) in general we have to be very careful when following a11y rules that were designed for desktop.

**Comment by aghassemi** [comment]
Also to update https://ampbyexample.com/advanced/tab_panels_with_amp-selector/preview/ sample to use this when ready (with `keyboard-support='select'`)

**Comment by kmh287** [comment]
@aghassemi Do we want to restrict the `select` option to single-select amp-selectors? I'm not sure how this would work with multi-select.

**Comment by aghassemi** [comment]
good point, yeah `select` does not make sense for multi-select case. If possible to restrict through validator, that would be great but I am not sure if validator support co-dependent rules. If not, just runtime `assert` should be good enough.

**Comment by kmh287** [comment]
Another edge case that comes up is initial selection with the `select` option. If the amp-selector starts with no initial selection and a user navigates to it with TAB, the first option will receive focus but won't be selected. In `select` mode, the user would need to navigate to another option and then navigate back to select the first option.
